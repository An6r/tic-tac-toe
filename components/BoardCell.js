import React from 'react';
import { PropTypes } from 'prop-types';
import cn from 'classnames';
import { SYMBOLS } from '../helpers/game';
import FontAwesome from 'react-fontawesome';

const BoardSpace = ({ column, index, value, handleCellClick, isWinnerMark, getPlayerSymbol }) => {
	const icon = (value) ? SYMBOLS[getPlayerSymbol(value)] : null;

	const cellClasses = cn('board-cell', {
		'winner-board-cell': (isWinnerMark(column, index))
	});

	return (
		<li className={cellClasses}
		  onClick={() => handleCellClick(column, index)}
		>
			{(value !== null) ?
				<FontAwesome name={icon} /> : null
			}
		</li>
	);
};

BoardSpace.propTypes = {
  column: PropTypes.number.isRequired,
  index: PropTypes.number.isRequired,
  value: PropTypes.number,
  handleCellClick: PropTypes.func.isRequired,
  /**
   * A function to check if the board space is in the winner marks.
   */
  isWinnerMark: PropTypes.func.isRequired,
  getPlayerSymbol: PropTypes.func.isRequired
};

export default BoardSpace;