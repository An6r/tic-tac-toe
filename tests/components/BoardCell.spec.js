import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import BoardCell from '../../components/BoardCell';

function setupComponent() {
  const props = {
    column: 2,
    index: 1,
    value: 1,
    handleCellClick: jest.fn(),
    isWinnerMark: jest.fn(),
    getPlayerSymbol: () => 'X',
  };

  const wrapper = shallow(
    <BoardCell {...props} />,
  );

  return {
    props,
    wrapper,
  };
}

describe('<BoardCell />', () => {
  it('should render', () => {
    const { wrapper } = setupComponent();

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should handle new sign on board space click', () => {
    const { wrapper, props } = setupComponent();

    wrapper.find('li').first().simulate('click');

    expect(props.handleCellClick.mock.calls).toEqual([
      [props.column, props.index],
    ]);
  });
});
