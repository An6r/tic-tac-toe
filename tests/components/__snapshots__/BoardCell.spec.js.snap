// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoardCell /> should render 1`] = `
<li
  className="board-cell"
  onClick={[Function]}
>
  <FontAwesome
    name="times"
  />
</li>
`;
